priv/repo/migrations/20211102141706_ensure_unique_active_section.exs

```diff
@@ -0,0 +1,8 @@
+defmodule Oli.Repo.Migrations.EnsureUniqueActiveSection do
+  use Ecto.Migration
+
+  def change do
+    # guarantee there is only ever one active section with a given context id
+    create unique_index(:sections, [:context_id], where: "status = 'active'", name: :sections_active_context_id_unique_index)
```

Comment:
How well will this work in an existing database like Proton?  Do we need to run some queries to ensure that Tokamak or Proton are not in a state already where it would be impossible to add this index?